src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -515,6 +520,33 @@ public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseReque
             requestData.Method = requestData.GetSingleHeaderValue(":method");
             requestData.Path = requestData.GetSingleHeaderValue(":path");
 
+            byte[]  body = null;
+            while ((frame.Flags & FrameFlags.EndStream) == 0)
+            {
+                frame = await ReadFrameAsync(Timeout).ConfigureAwait(false);
+                // TODO. support more types.
+                Assert.Equal(FrameType.Data, frame.Type);
+                if (frame.Type == FrameType.Data && frame.Length > 1)
+                {
+                    if (body == null)
+                    {
+                        body = ((DataFrame)frame).Data.ToArray();
```

Comment:
For code clarity, it seems better to cast the frame to a DataFrame once instead of doing it every time you use it